En esta jupyter notebook se desarrolla la extracción, transformación y carga del conjunto de datos para 2018.

In [1]:
import pandas as pd
import json

%load_ext autoreload
%autoreload 2

import utils

import warnings
warnings.filterwarnings("ignore")

Se extraen los datos desde el archivo .csv, se convierte en Dataframe y se observa su contenido.

In [2]:
# Ruta al dataset
ruta_siniestro2018 = '../Datasets/feu-siniestros - (FC)feu-siniestros-2018.csv'

# Leer el archivo CSV directamente en un DataFrame
df_sin2018 = pd.read_csv(ruta_siniestro2018)

# Mostrar el DataFrame
df_sin2018

id_feu    unidad_regional siniestro_fecha siniestro_hora   provincia  \
0     49056  OBSERVATORIO VIAL      2018-10-29       20:10:00  CORRIENTES   
1     49059  OBSERVATORIO VIAL      2018-10-31       19:30:00  CORRIENTES   
2     48998  OBSERVATORIO VIAL      2018-10-08       13:10:00  CORRIENTES   
3     49069  OBSERVATORIO VIAL      2018-11-05        7:15:00  CORRIENTES   
4     48869  OBSERVATORIO VIAL      2018-08-25       10:30:00  CORRIENTES   
..      ...                ...             ...            ...         ...   
401   49032  OBSERVATORIO VIAL      2018-10-21       19:10:00  CORRIENTES   
402   49118  OBSERVATORIO VIAL      2018-11-28       14:20:00  CORRIENTES   
403   49160  OBSERVATORIO VIAL      2018-12-11       21:00:00  CORRIENTES   
404   49108  OBSERVATORIO VIAL      2018-11-22       22:50:00  CORRIENTES   
405   49009  OBSERVATORIO VIAL      2018-10-12        9:20:00  CORRIENTES   

     id_provincia departamento  id_departamento   localidad zona_ocurrencia  \
0              18      CAPITAL             21.0  CORRIENTES          Urbana   
1              18      CAPITAL             21.0  CORRIENTES          Urbana   
2              18      CAPITAL             21.0  CORRIENTES          Urbana   
3              18      CAPITAL             21.0  CORRIENTES          Urbana   
4              18      CAPITAL             21.0  CORRIENTES          Urbana   
..            ...          ...              ...         ...             ...   
401            18      CAPITAL             21.0  CORRIENTES          Urbana   
402            18      CAPITAL             21.0  CORRIENTES          Urbana   
403            18      CAPITAL             21.0  CORRIENTES          Urbana   
404            18      CAPITAL             21.0  CORRIENTES          Urbana   
405            18      CAPITAL             21.0  CORRIENTES          Urbana   

     ... senalizacion_carteleria senalizacion_horizontal  \
0    ...                      Si                      Si   
1    ...                      No                      No   
2    ...                      No                      No   
3    ...                      Si                      Si   
4    ...                      No                      Si   
..   ...                     ...                     ...   
401  ...                      No                      No   
402  ...                      No                      No   
403  ...                      Si                      Si   
404  ...                      No                      No   
405  ...                      Si                      Si   

    senalizacion_transitoria luminosidad luz_artificial transito_restringido  \
0                         No   Atardecer            NaN                  NaN   
1                         No       Noche             NO      Sin Restricción   
2                         No         Día             NO                  NaN   
3                        S/D         Día             NO      Sin Restricción   
4                        NaN         Día             NO                  NaN   
..                       ...         ...            ...                  ...   
401                       No       Noche             SI      Sin Restricción   
402                       No         Día             NO      Sin Restricción   
403                       No       Noche             SI      Sin Restricción   
404                       No       Noche             SI      Sin Restricción   
405                       No         Día             NO      Sin Restricción   

    estado_ambiental  visibilidad  visibilidad_otro      semaforo  
0          Despejado        Buena               NaN      Funciona  
1            Nublado        Buena               NaN  Sin Semáforo  
2          Despejado        Buena               NaN  Sin Semáforo  
3          Despejado        Buena               NaN      Funciona  
4          Despejado        Buena               NaN  Sin Semáforo  
..               ...          ...        

Se revisan los tipos de datos por columna y la cantidad de nulos.

In [3]:
utils.verificar_tipo_datos(df_sin2018)

nombre_campo                        tipo_datos  \
0                              id_feu                   [<class 'int'>]   
1                     unidad_regional                   [<class 'str'>]   
2                     siniestro_fecha                   [<class 'str'>]   
3                      siniestro_hora                   [<class 'str'>]   
4                           provincia                   [<class 'str'>]   
5                        id_provincia                   [<class 'int'>]   
6                        departamento  [<class 'str'>, <class 'float'>]   
7                     id_departamento                 [<class 'float'>]   
8                           localidad                   [<class 'str'>]   
9                     zona_ocurrencia                   [<class 'str'>]   
10                        via_publica                   [<class 'str'>]   
11                         nombre_via                   [<class 'str'>]   
12                          altura_km                   [<class 'str'>]   
13                      entre_calle_1  [<class 'str'>, <class 'float'>]   
14                      entre_calle_2  [<class 'float'>, <class 'str'>]   
15                            latitud  [<class 'str'>, <class 'float'>]   
16                           longitud  [<class 'str'>, <class 'float'>]   
17                             ilesos                   [<class 'int'>]   
18                            heridos                   [<class 'int'>]   
19                         fallecidos                   [<class 'int'>]   
20                          vehiculos                   [<class 'int'>]   
21                           peatones                   [<class 'int'>]   
22            tipo_siniestro_multiple  [<class 'str'>, <class 'float'>]   
23               tipo_siniestro_unico  [<class 'str'>, <class 'float'>]   
24                    despiste_previo                   [<class 'str'>]   
25                tipo_siniestro_otro  [<class 'float'>, <class 'str'>]   
26                      tipo_colision  [<class 'str'>, <class 'float'>]   
27           cantidad_de_involucrados                   [<class 'int'>]   
28                        trazado_via                   [<class 'str'>]   
29            configuracion_de_la_via  [<class 'str'>, <class 'float'>]   
30  configuracion_de_la_via_otro_tipo  [<class 'float'>, <class 'str'>]   
31                  pendiente_declive  [<class 'str'>, <class 'float'>]   
32                     cono_de_vision                 [<class 'float'>]   
33             material_de_la_calzada  [<class 'str'>, <class 'float'>]   
34        material_de_la_calzada_otro                 [<class 'float'>]   
35               estado_de_la_calzada  [<class 'str'>, <class 'float'>]   
36            estado_fisico_ambiental  [<class 'str'>, <class 'float'>]   
37        estado_de_conservacion_otro                 [<class 'float'>]   
38          division_fisica_de_la_via  [<class 'str'>, <class 'float'>]   
39     division_fisica_de_la_via_otro  [<class 'float'>, <class 'str'>]   
40              barreras_de_seguridad                 [<class 'float'>]   
41                          banquinas                 [<class 'float'>]   
42            senalizacion_carteleria  [<class 'str'>, <class 'float'>]   
43            senalizacion_horizontal  [<class 'str'>, <class 'float'>]   
44           senalizacion_transitoria  [<class 'str'>, <class 'float'>]   
45                        luminosidad  [<class 'str'>, <class 'float'>]   
46                     luz_artificial  [<class 'float'>, <class 'str'>]   
47               transito_restringido  [<class 'float'>, <class 'str'>]   
48                   estado_ambiental  [<class 'str'>, <class 'float'>]   
49                        visibilidad  [<class 'str'>, <class 'float'>]   
50                   visibilidad_otro  [<class 'float'>, <class 'str'>]   
51                           semaforo  [<class 'str'>, <class 'float'>]   

    no_nulos_%  nulos_%  nulos  
0       100.00  

Se observa que hay columnas con más de 90% de nulos, por lo cual decidimos borrarlas.

In [4]:
# Eliminar columnas con más del 50% de valores nulos
umbral = 0.8  # Umbral del 50%
df_2018filtrado = df_sin2018.dropna(thresh=len(df_sin2018) * (1 - umbral), axis=1)

# Mostrar las columnas que se mantuvieron
df_2018filtrado.columns

Index(['id_feu', 'unidad_regional', 'siniestro_fecha', 'siniestro_hora',
       'provincia', 'id_provincia', 'departamento', 'id_departamento',
       'localidad', 'zona_ocurrencia', 'via_publica', 'nombre_via',
       'altura_km', 'entre_calle_1', 'latitud', 'longitud', 'ilesos',
       'heridos', 'fallecidos', 'vehiculos', 'peatones',
       'tipo_siniestro_multiple', 'tipo_siniestro_unico', 'despiste_previo',
       'tipo_colision', 'cantidad_de_involucrados', 'trazado_via',
       'configuracion_de_la_via', 'pendiente_declive',
       'material_de_la_calzada', 'estado_de_la_calzada',
       'estado_fisico_ambiental', 'division_fisica_de_la_via',
       'senalizacion_carteleria', 'senalizacion_horizontal',
       'senalizacion_transitoria', 'luminosidad', 'luz_artificial',
       'transito_restringido', 'estado_ambiental', 'visibilidad', 'semaforo'],
      dtype='object')

Verificamos ahora duplicados

In [5]:
utils.verifica_duplicados_por_columna(df_2018filtrado, 'id_feu')

'No hay duplicados'

No hay duplicados, podemos continuar corrigiendo los tipos de datos en algunas columnas. Por ejemplo, fechas y horas.

In [6]:
# Convertir la columna siniestro_fecha a tipo datetime
df_2018filtrado['siniestro_fecha'] = pd.to_datetime(df_2018filtrado['siniestro_fecha'], format='%Y-%m-%d', errors='coerce')

# Convertir la columna siniestro_hora a tipo time (si está en formato 24 horas)
df_2018filtrado['siniestro_hora'] = pd.to_datetime(df_2018filtrado['siniestro_hora'], format='%H:%M:%S', errors='coerce').dt.time

# Verificar las conversiones
df_2018filtrado[['siniestro_fecha', 'siniestro_hora']].head()


siniestro_fecha siniestro_hora
0      2018-10-29       20:10:00
1      2018-10-31       19:30:00
2      2018-10-08       13:10:00
3      2018-11-05       07:15:00
4      2018-08-25       10:30:00

Ahora, podemos pasar en limpio qué datos contiene este Dataframe

In [7]:
utils.verificar_tipo_datos(df_2018filtrado)

nombre_campo  \
0                      id_feu   
1             unidad_regional   
2             siniestro_fecha   
3              siniestro_hora   
4                   provincia   
5                id_provincia   
6                departamento   
7             id_departamento   
8                   localidad   
9             zona_ocurrencia   
10                via_publica   
11                 nombre_via   
12                  altura_km   
13              entre_calle_1   
14                    latitud   
15                   longitud   
16                     ilesos   
17                    heridos   
18                 fallecidos   
19                  vehiculos   
20                   peatones   
21    tipo_siniestro_multiple   
22       tipo_siniestro_unico   
23            despiste_previo   
24              tipo_colision   
25   cantidad_de_involucrados   
26                trazado_via   
27    configuracion_de_la_via   
28          pendiente_declive   
29     material_de_la_calzada   
30       estado_de_la_calzada   
31    estado_fisico_ambiental   
32  division_fisica_de_la_via   
33    senalizacion_carteleria   
34    senalizacion_horizontal   
35   senalizacion_transitoria   
36                luminosidad   
37             luz_artificial   
38       transito_restringido   
39           estado_ambiental   
40                visibilidad   
41                   semaforo   

                                           tipo_datos  no_nulos_%  nulos_%  \
0                                     [<class 'int'>]      100.00     0.00   
1                                     [<class 'str'>]      100.00     0.00   
2   [<class 'pandas._libs.tslibs.timestamps.Timest...      100.00     0.00   
3                           [<class 'datetime.time'>]      100.00     0.00   
4                                     [<class 'str'>]      100.00     0.00   
5                                     [<class 'int'>]      100.00     0.00   
6                    [<class 'str'>, <class 'float'>]       99.51     0.49   
7                                   [<class 'float'>]       99.51     0.49   
8                                     [<class 'str'>]      100.00     0.00   
9                                     [<class 'str'>]      100.00     0.00   
10                                    [<class 'str'>]      100.00     0.00   
11                                    [<class 'str'>]      100.00     0.00   
12                                    [<class 'str'>]      100.00     0.00   
13                   [<class 'str'>, <class 'float'>]       76.60    23.40   
14                   [<class 'str'>, <class 'float'>]       98.77     1.23   
15                   [<class 'str'>, <class 'float'>]       98.77     1.23   
16                                    [<class 'int'>]      100.00     0.00   
17                                    [<class 'int'>]      100.00     0.00   
18                                    [<class 'int'>]      100.00     0.00   
19                                    [<class 'int'>]      100.00     0.00   
20                                    [<class 'int'>]      100.00     0.00   
21                   [<class 'str'>, <class 'float'>]       99.01     0.99   
22                   [<class 'str'>, <class 'float'>]       99.01     0.99   
23                                    [<class 'str'>]      100.00     0.00   
24                   [<class 'str'>, <class 'float'>]       72.41    27.59   
25                                    [<class 'int'>]      100.00     0.00   
26                                    [<class 'str'>]      100.00     0.00   
27                   [<class 'str'>, <class 'float'>]       83.25    16.75   
28                   [<class 'str'>, <class 'float'>]       93.60     6.40   
29                   [<class 'str'>, <class 'float'>]       98.28     1.72   
30                   [<class 'str'>, <class 'float'>]       95.32     4.68   
31                   [<class 'str'>, <class 'float'>]       87.93    12.07   
32                   [<

In [8]:
print(df_2018filtrado.shape)

(406, 42)


Este conjunto contiene 42 columnas y quedó con 406 filas luego de borrar las columnas con mayor porcentaje de nulos. Las columnas que contiene son:

* id_feu: es el índice del siniestro.
* unidad_regional: es el nombre de la unidad regional encargada de cargar el dato. En este caso, el 100% corresponde a Observatorio Vial.
* siniestro_fecha: es la fecha del siniestro.
* siniestro_hora: es la hora del siniestro.
* id_provincia: es la provincia en la que ocurrió el siniestro. En nuestro caso, el 100% corresponde a Corrientes.
* departamento: es el departamento en el que ocurrió el siniestro. En nuestro caso, el 100% corresponde a Capital.
* id_departamento: es la ID del departamento en el que ocurrió el siniestro.
* localidad: es la localidad en la que ocurrió el siniestro. En nuestro caso, el 100% corresponde a Corrientes.
* zona_ocurrencia: es la especificación sobre el tipo de zona donde ocurrió el siniestro. Urbana o Rural.
* via_publica: es la especificación sobre el tipo de calle en la que ocurrió el siniestro. Avenida, Calle o Ruta.
* nombre_via: es el nombre de la calle, avenida o ruta donde ocurrió el siniestro.
* altura_km: es la altura en kilómetros dónde ocurrió el siniestro.
* entre_calle_1: es el cruce en donde ocurrió el siniestro.
* latitud: es la latitud de dónde ocurrió el siniestro.
* longitud: es la longitud de dónde ocurrió el siniestro.
* ilesos: es la cantidad de personas que no sufrieron daños en el siniestro.
* heridos: es la cantidad de personas que sufrieron daños en el siniestro.
* fallecidos: es la cantidad de personas que fallecieron en el siniestro.
* vehiculos: es la cantidad de vehículos involucrados en el siniestro.
* peatones: es la cantidad de peatones involucrados en el siniestro.
* tipo_siniestro_multiple: es la especificación del hecho.
* despiste_previo: es la respuesta ante el hecho.
* tipo_colision: es el tipo de colisión del siniestro.
* cantidad_involucrados: es la cantidad de personas que participaron del siniestro.
* trazado_via: es la especficidad sobre el tipo de calle en que ocurrió el hecho. Recta o Curva.
* configuracion_de_la_via: es el tipo de configuración de la vía. Intersección, Cruce, Rotonda.
* pendiente_declive: es si existe pendiente.
* material_de_la_calzada: es el material del que está hecha la calzada donde ocurrió el siniestro.
* estado_de_la_calzada: es el estado en que se encontraba la calzada en el momento del siniestro.
* estado_fisico_ambiental: el estado de la calzada al momento del siniestro. Seca, Mojada.
* division_fisica_de_la_via: es el qué divide la vía donde ocurrió el siniestro. Por ejemplo, barrera de hormigón.
* senalización_carteleria: es si existía señalización en la zona al momento del hecho.
* senalización_horizontal: es si existía señalización horizontal en la zona al momento del hecho.
* senalización_transitoria: es si existía señalización transitoria en la zona al momento del hecho.
* luminosidad: es si el hecho ocurrió de día o de noche. Luz Natural.
* luz_artifical: es si en el lugar del hecho existía luz artificial.
* transito_restringido: es si el lugar era de tránsito restringido.
* estado_ambiental: es cómo estaba el tiempo al momento del hecho. Despejado, lluvioso, etc.
* visibilidad: es cómo se encontraba la visibilidad al momento del hecho. 
* semaforo: es si existe semáforo en la zona del hecho, y su funcionamiento.

In [9]:
# Eliminar columnas innecesarias
columnas_a_eliminar = [
    'unidad_regional',
    'id_provincia',
    'id_departamento',
    'altura_km',
    'tipo_siniestro_multiple',
    'despiste_previo',
    'tipo_colision',
    'trazado_via',
    'configuracion_de_la_via',
    'pendiente_declive',
    'division_fisica_de_la_via',
    'senalizacion_carteleria',
    'senalizacion_horizontal',
    'senalizacion_transitoria',
    'transito_restringido', 
    'provincia',
    'departamento',
    'localidad',
    ]

df_2018filtrado = df_2018filtrado.drop(columns=columnas_a_eliminar)


Ahora tenemos un Dataframe mucho más manejable, con los datos más importantes para nuestra investigación. Vamos a guardarlo para su futuro uso.

In [10]:
sin2018 = '../Datasets_limpios/siniestros2018.csv'
df_2018filtrado.to_csv(sin2018)

A continuación, veremos qué contiene el segundo Datasets que tenemos para el año 2018.

In [11]:
# Ruta al dataset
ruta_involucrados2018 = '../Datasets/feu-involucrado-descarga - (FC)feu-involucrado-2018.csv'

# Leer el archivo CSV directamente en un DataFrame
df_inv2018 = pd.read_csv(ruta_involucrados2018)

# Mostrar el DataFrame
df_inv2018

id_feu      siniestro_fecha   provincia  id_provincia  numero  \
0     48796   2018-08-14 7:30:00  CORRIENTES            21       0   
1     48796   2018-08-14 7:30:00  CORRIENTES            21       1   
2     48823  2018-08-08 15:00:00  CORRIENTES            21       0   
3     48823  2018-08-08 15:00:00  CORRIENTES            21       1   
4     48832  2018-08-10 14:10:00  CORRIENTES            21       0   
..      ...                  ...         ...           ...     ...   
786   49160  2018-12-11 21:00:00  CORRIENTES            21       1   
787   49004  2018-10-09 19:00:00  CORRIENTES            21       1   
788   48997  2018-10-07 11:40:00  CORRIENTES            21       0   
789   48982  2018-10-01 18:15:00  CORRIENTES            21       1   
790   48904  2018-09-04 10:30:00  CORRIENTES            21       1   

     numero_ocupantes tipo_involucrado  otro_tipo_involucrado  ano_vehiculo  \
0                   1        Automóvil                    NaN        2007.0   
1                   1        Automóvil                    NaN        2010.0   
2                   1        Automóvil                    NaN        2000.0   
3                   1        Automóvil                    NaN        2015.0   
4                   1        Automóvil                    NaN        2016.0   
..                ...              ...                    ...           ...   
786                 1           Peatón                    NaN           NaN   
787                 1           Peatón                    NaN           NaN   
788                 1           Peatón                    NaN           NaN   
789                 1           Peatón                    NaN           NaN   
790                 1           Peatón                    NaN           NaN   

    uso_vehiculo tipo_servicio_vehiculo  tipo_vehiculo_de_movilidad_personal  \
0        Privado                    NaN                                  NaN   
1        Privado                    NaN                                  NaN   
2            NaN                    NaN                                  NaN   
3        Privado                    NaN                                  NaN   
4        Privado                    NaN                                  NaN   
..           ...                    ...                                  ...   
786          NaN                    NaN                                  NaN   
787          NaN                    NaN                                  NaN   
788          NaN                    NaN                                  NaN   
789          NaN                    NaN                                  NaN   
790          NaN                    NaN                                  NaN   

     tipo_de_bicicleta  tipo_de_transporte_de_carga  sustancias_peligrosas  \
0                  NaN                          NaN                    NaN   
1                  NaN                          NaN                    NaN   
2                  NaN                          NaN                    NaN   
3                  NaN                          NaN                    NaN   
4                  NaN                          NaN                    NaN   
..                 ...                          ...                    ...   
786                NaN                          NaN                    NaN   
787                NaN                          NaN                    NaN   
788                NaN                          NaN                    NaN   
789                NaN                          NaN                    NaN   
790                NaN                          NaN                    NaN   

     tipo_transporte_pasajeros  tipo_traccion_a_sangre  
0                          NaN                     NaN  
1                          NaN                     NaN  
2                          NaN                     NaN  
3                          NaN                     NaN  
4                          NaN  

No parece un Dataset muy valioso. Vamos a confirmarlo.


In [12]:
utils.verificar_tipo_datos(df_inv2018)

nombre_campo                        tipo_datos  \
0                                id_feu                   [<class 'int'>]   
1                       siniestro_fecha                   [<class 'str'>]   
2                             provincia                   [<class 'str'>]   
3                          id_provincia                   [<class 'int'>]   
4                                numero                   [<class 'int'>]   
5                      numero_ocupantes                   [<class 'int'>]   
6                      tipo_involucrado                   [<class 'str'>]   
7                 otro_tipo_involucrado                 [<class 'float'>]   
8                          ano_vehiculo                 [<class 'float'>]   
9                          uso_vehiculo  [<class 'str'>, <class 'float'>]   
10               tipo_servicio_vehiculo  [<class 'float'>, <class 'str'>]   
11  tipo_vehiculo_de_movilidad_personal                 [<class 'float'>]   
12                    tipo_de_bicicleta                 [<class 'float'>]   
13          tipo_de_transporte_de_carga                 [<class 'float'>]   
14                sustancias_peligrosas                 [<class 'float'>]   
15            tipo_transporte_pasajeros                 [<class 'float'>]   
16               tipo_traccion_a_sangre                 [<class 'float'>]   

    no_nulos_%  nulos_%  nulos  
0       100.00     0.00      0  
1       100.00     0.00      0  
2       100.00     0.00      0  
3       100.00     0.00      0  
4       100.00     0.00      0  
5       100.00     0.00      0  
6       100.00     0.00      0  
7         0.00   100.00    791  
8        96.46     3.54     28  
9        85.21    14.79    117  
10        7.59    92.41    731  
11        0.00   100.00    791  
12        0.00   100.00    791  
13        0.00   100.00    791  
14        0.00   100.00    791  
15        0.00   100.00    791  
16        0.00   100.00    791

La mayoría de estos datos se encuentran nulos, o no nos agregan nada a nuestro futuro EDA. Por lo cual, se decide NO utilizarlos.

Procedemos a analizar entonces, el último posible dataset a utilizar. Semáforos.

In [13]:
# Ruta al dataset
ruta_semaforos = '../Datasets/bdgis_semaforos - recurso_semaforos (1).csv'

# Leer el archivo CSV directamente en un DataFrame
df_semaforos = pd.read_csv(ruta_semaforos)

# Mostrar el DataFrame
df_semaforos

gid nom_barrio  esquina                            calle  \
0      1        NaN      1.0                   AV. 3 DE ABRIL   
1      2        NaN      2.0                   AV. 3 DE ABRIL   
2      3        NaN      3.0                   AV. 3 DE ABRIL   
3      4        NaN      4.0                   AV. 3 DE ABRIL   
4      5        NaN      5.0                   AV. 3 DE ABRIL   
..   ...        ...      ...                              ...   
382  295        NaN    295.0  PDTE. DR. RAUL RICARDO ALFONSIN   
383  296        NaN    296.0  PDTE. DR. RAUL RICARDO ALFONSIN   
384  508        NaN      NaN                   AV. 3 DE ABRIL   
385   96        NaN     96.0                AV. INDEPENDENCIA   
386  559     CENTRO    193.0                   PEATONAL JUNIN   

                             calle2          suj-column  nro_artef  \
0    AV. COSTANERA GRAL. SAN MARTIN  con pescante 5,5m.        3.0   
1    AV. COSTANERA GRAL. SAN MARTIN  con pescante 5,5m.        2.0   
2    AV. COSTANERA GRAL. SAN MARTIN  con pescante 5,5m.        4.0   
3                             CHACO  con pescante 5,5m.        4.0   
4                             CHACO  con pescante 5,5m.        2.0   
..                              ...                 ...        ...   
382                      GODOY CRUZ  con pescante 5,5m.        2.0   
383                      GODOY CRUZ  con pescante 5,5m.        2.0   
384                       CATAMARCA                 NaN        NaN   
385                   AV. CHACABUCO  con pescante 5,5m.        2.0   
386                         MENDOZA                 NaN        2.0   

       sem-artef   sem-arte_2 sem-arte_3  ...           led tipo led  \
0    1*300+2*200        3*200    3*200\n  ...  amarillo 200      12v   
1    1*300+2*200        3*200    ninguno  ...  amarillo 200      12v   
2    1*300+2*200        3*200    3*200\n  ...  amarillo 200      12v   
3    1*300+2*200  1*300+2*200    3*200\n  ...  amarillo 200      12v   
4          3*300        3*300    ninguno  ...  amarillo 200      12v   
..           ...          ...        ...  ...           ...      ...   
382  1*300+2*200        3*200    ninguno  ...  amarillo 200      12v   
383  1*300+2*200        3*200    ninguno  ...  amarillo 200      12v   
384          NaN          NaN        NaN  ...           NaN      NaN   
385        3*200        3*300    ninguno  ...  amarillo 200      12v   
386          NaN          NaN        NaN  ...           NaN      NaN   

    suj-soport de-cargado de-bateria de-cabl al de_cam s_n c-estado  \
0        doble        24V         no      bueno         no    bueno   
1       simple        24V         no      bueno         no    bueno   
2        doble    Ninguno         no      bueno         no    bueno   
3        doble        12V         no      bueno         no    bueno   
4        doble        24V         no      bueno         no    bueno   
..         ...        ...        ...        ...        ...      ...   
382      doble    Ninguno         no      bueno         no    bueno   
383      doble    Ninguno         no      bueno         no    bueno   
384        NaN        NaN        NaN        NaN        NaN      NaN   
385      doble        12V         no      bueno         no    bueno   
386        NaN        NaN        NaN        NaN        NaN      NaN   

           lng        lat  
0   -58.855530 -27.473426  
1   -58.855712 -27.473398  
2   -58.855479 -27.473166  
3   -58.847385 -27.474526  
4   -58.847295 -27.474420  
..         ...        ...  
382 -58.833872 -27.476002  
383 -58.814533 -27.478528  
384 -58.835615 -27.467602  
385        NaN        NaN  
386        NaN        NaN  

[387 rows x 26 columns]

Nuevamente los datos no son de vital importancia para nuestro proyecto, sin embargo, nos quedaremos con algunas columnas importantes.

In [14]:
df_semaforos = df_semaforos[['lat', 'lng']]

df_semaforos.head()

lat        lng
0 -27.473426 -58.855530
1 -27.473398 -58.855712
2 -27.473166 -58.855479
3 -27.474526 -58.847385
4 -27.474420 -58.847295

Guardamos este primer nuevo Dataset para su uso en la posteridad. 

In [15]:
output_path = '../Datasets_limpios/semaforos_lat_lng.csv'
df_semaforos.to_csv(output_path)